src/Apps/Order2/Routes/Checkout/Components/Order2PaymentStep.tsx

```diff
@@ -0,0 +1,125 @@
+import CheckmarkIcon from "@artsy/icons/CheckmarkIcon"
+import { Clickable, Flex, Spacer, Text } from "@artsy/palette"
+import { Order2PaymentForm } from "Apps/Order2/Routes/Checkout/Components/Order2PaymentForm"
+import { Brand, BrandCreditCardIcon } from "Components/BrandCreditCardIcon"
+import type { Order2PaymentForm_order$key } from "__generated__/Order2PaymentForm_order.graphql"
+import { useState } from "react"
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { graphql } from "react-relay"
+import { Order2PaymentStepConfirmationTokenQuery } from "__generated__/Order2PaymentStepConfirmationTokenQuery.graphql"
+
+interface Order2PaymentStepProps {
+  order: Order2PaymentForm_order$key
+}
+
+export const Order2PaymentStep: React.FC<Order2PaymentStepProps> = ({
+  order,
+}) => {
+  // Temporarily using a local state until we have a global checkout context
+  const [confirmationTokenID, setConfirmationTokenID] = useState<string | null>(
+    null,
+  )
+  const paymentStepComplete = !!confirmationTokenID
+
+  return (
+    <>
+      {!paymentStepComplete && (
+        <Flex flexDirection="column" backgroundColor="mono0" p={2}>
+          <Text variant="sm-display" fontWeight={500} color="mono100">
+            Payment
+          </Text>
+          <Text variant="xs" color="mono60">
+            Options vary based on price, gallery, and location
+          </Text>
+          <Spacer y={2} />
+          <Order2PaymentForm
+            setConfirmationTokenID={setConfirmationTokenID}
+            order={order}
+          />
+        </Flex>
+      )}
+
+      {paymentStepComplete && (
+        <MeConfirmationTokenQueryRenderer
+          tokenID={confirmationTokenID}
+          setConfirmationTokenID={setConfirmationTokenID}
+        />
+      )}
+    </>
```

Comment:
Instead of mounting and unmounting components as the boolean value changes, we may want to use a pattern like `<Box display={paymentStepComplete ?  "block" : "none"}><MeConfirmationToken...`. This could be on a case by case basis depending on whether the mount causes side effects or is costly.